src/data/padics/padic_integers.lean

```diff
@@ -50,29 +50,38 @@ namespace padic_int
 variables {p : ℕ} [fact p.prime]
 
 /-- Addition on ℤ_p is inherited from ℚ_p. -/
-def add : ℤ_[p] → ℤ_[p] → ℤ_[p]
-| ⟨x, hx⟩ ⟨y, hy⟩ := ⟨x+y,
-    le_trans (padic_norm_e.nonarchimedean _ _) (max_le_iff.2 ⟨hx,hy⟩)⟩
+instance : has_add ℤ_[p] :=
+⟨λ ⟨x, hx⟩ ⟨y, hy⟩, ⟨x+y,
+    le_trans (padic_norm_e.nonarchimedean _ _) (max_le_iff.2 ⟨hx,hy⟩)⟩⟩
+
+@[simp, nolint def_lemma, nolint doc_blame, _refl_lemma]
+def add_mk (x y hx hy) : @has_add.add ℤ_[p] _ ⟨x, hx⟩ ⟨y, hy⟩ = ⟨x + y, _⟩ := rfl
```

Comment:
Can you add an explanation of this change (and the other non-obvious ones) to the commit message?

Reply:
Ok, I agree that three declarations with two nolints each are a bit excessive.  I'll clean this up.

I did this to make the ring instance below work.  There we want to unfold addition for padic integers but not other additions, so you can't say `simp [(+)]`.

Reply:
Thanks, this is great!